File: project/backend-services/sharespot-device-records-slave-backend/domain/src/main/java/sharespot/services/devicerecordsbackend/domain/model/Device.java
package sharespot.services.devicerecordsbackend.domain.model;

import sharespot.services.devicerecordsbackend.domain.model.records.DeviceId;
import sharespot.services.devicerecordsbackend.domain.model.records.DeviceName;

public record Device(DeviceId id, DeviceName name) {
}

File: project/frontend-services/libs/smart-irrigation/model/src/lib/model/device/Device.ts
import {DeviceType} from "./DeviceType";
import {RecordEntry} from "./RecordEntry";
import {DeviceId} from "./DeviceId";
import {DeviceName} from "./DeviceName";

export class Device {
  constructor(public id: DeviceId, public type: DeviceType, public name: DeviceName, public records: Array<RecordEntry>, public remoteControl: boolean) {

  }
}
